### PR TITLE
Use TanStack Table instead of self-implemented DataGrid

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

NA

## What does this implement/fix? Explain your changes.

Use TanStack Table instead of self-implemented DataGrid

### CODE REVIEW

File: optuna_dashboard/ts/components/DataGrid.tsx [hunk 1]
```diff
@@ -257,202 +58,312 @@ const HiddenSpan = styled("span")({
...
+    onPaginationChange: setPagination,
+    getCoreRowModel: getCoreRowModel(),
+    getFacetedRowModel: getFacetedRowModel(),
+    getFacetedUniqueValues: getFacetedUniqueValues(),
+    getSortedRowModel: getSortedRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    autoResetPageIndex: false,
+    //
+    // debugTable: true,
```
Comment (keisuke-umezawa) [inline optuna_dashboard/ts/components/DataGrid.tsx:96]: [todo] Need to delete.

**Comment by c-bata** [comment]
@porink0424 Could you review this PR?

File: optuna_dashboard/ts/components/Preferential/PreferentialAnalytics.tsx [hunk 2]
```diff
@@ -8,21 +8,35 @@ import {
...
   const userAttrs = studySummary?.user_attrs || studyDetail?.user_attrs || []
-  const userAttrColumns: DataGridColumn<Optuna.Attribute>[] = [
-    { field: "key", label: "Key", sortable: true },
-    { field: "value", label: "Value", sortable: true },
+  const columnHelper = createColumnHelper<Optuna.Attribute>()
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const columns: ColumnDef<Optuna.Attribute, any>[] = [
+    columnHelper.accessor("key", {
+      header: "Key",
+      footer: (info) => info.column.id,
```
Comment (porink0424) [inline optuna_dashboard/ts/components/Preferential/PreferentialAnalytics.tsx:30]: [nits]: Footer is not created in the table, is this prop really necessary or not?

**Review by porink0424 — COMMENTED** [review]
Looks almost great, but I left some nits!

**Comment by keisuke-umezawa** [comment]
@porink0424 
I followed your comments and rebased with main. Could you review it again?

**Review by porink0424 — APPROVED** [review]
LGTM👍

**Pull request merged**